components/brave_ads/browser/bundle_state_database.cc

```diff
@@ -398,54 +402,67 @@ bool BundleStateDatabase::InsertOrUpdateAdInfoCategory(
           "(ad_info_uuid, category_name) "
           "VALUES (?, ?)"));
 
-  ad_info_statement.BindString(0, ad_info.uuid);
+  ad_info_statement.BindString(0, info.creative_instance_id);
   ad_info_statement.BindString(1, category);
 
   return ad_info_statement.Run();
 }
 
-bool BundleStateDatabase::GetAdsForCategory(
-    const std::string& category,
-    std::vector<ads::AdInfo>* ads) {
+bool BundleStateDatabase::GetCreativeAdNotifications(
+    const std::vector<std::string>& categories,
+    ads::CreativeAdNotificationList* ads) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
 
+  DCHECK(ads);
+  if (!ads) {
+    return false;
+  }
+
```

Comment:
nit: as discussed, Chromium standards say do not DCHECK and then guard... so remove the if? Even if I still don't agree with it... 😄